test/t8_IO/t8_gtest_vtk_write_read.cxx

```diff
@@ -0,0 +1,95 @@
+/*
+  This file is part of t8code.
+  t8code is a C library to manage a collection (a forest) of multiple
+  connected adaptive space-trees of general element classes in parallel.
+
+  Copyright (C) 2024 the developers
+
+  t8code is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2 of the License, or
+  (at your option) any later version.
+
+  t8code is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with t8code; if not, write to the Free Software Foundation, Inc.,
+  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*/
+
+#include <gtest/gtest.h>
+#include <t8_vtk/t8_vtk_writer.hxx>
+#include <t8_cmesh.h>
+#include <t8_cmesh/t8_cmesh_trees.h>
+#include <t8_cmesh/t8_cmesh_examples.h>
+#include <t8_schemes/t8_default/t8_default.hxx>
+#include "test/t8_cmesh_generator/t8_cmesh_example_sets.hxx"
+#include <t8_cmesh/t8_cmesh_partition.h>
+
+#include <t8_vtk/t8_vtk_writer.h>
+#include <t8_vtk/t8_vtk_reader.hxx>
+
+class vtk_write_read_test: public testing::TestWithParam<cmesh_example_base *> {
+ protected:
+  void
+  SetUp () override
+  {
+    cmesh = GetParam ()->cmesh_create (); /* create a cmesh from example base */
+  }
+
+  void
+  TearDown () override
+  {
+    std::remove ("test_vtk_0.vtu");
+    std::remove ("test_vtk.pvtu"); /* delete files after test finished */
```

Comment:
```suggestion
    std::remove ("test_vtk.pvtu"); /* delete files after test finished */
    t8_cmesh_unref (&cmesh);
```

It is better to unref the cmesh in the tear-down part. That way the cmesh is deleted even if the test fails at some point.